### PR TITLE
Cleanup `BoyerMoore`

### PR DESCRIPTION
**Pull request by prathameshpowar1910**
- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized it.
- [ ] All filenames are in PascalCase.
- [ ] All functions and variable names follow Java naming conventions.
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [ ] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Review by vil02 — CHANGES_REQUESTED** [review]
Please split the `findmajor` method as described in the issue. I think it is also worth to rename it to `findMajor`.

I have pushed one commit into your branch: fe687c37c5ddc0421865a3cfe8ccfca8d515127b. The Most important change is to
```java
return Optional.of(cand);
```

Besides that I have adapted the tests.

This PR closes #4949.

File: src/main/java/com/thealgorithms/others/BoyerMoore.java [hunk 1]
```diff
@@ -5,35 +5,51 @@
...
+import java.util.Optional;
 
 public final class BoyerMoore {
     private BoyerMoore() {
     }
 
-    public static int findmajor(final int[] a) {
+    public static Optional<Integer> findMajor(final int[] a) {
+        int candidate = findCandidate(a);
+        int count = countOccurrences(a, candidate);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/others/BoyerMoore.java:16]:
```suggestion
        final var candidate = findCandidate(a);
        final var count = countOccurrences(a, candidate);
```

File: src/main/java/com/thealgorithms/others/BoyerMoore.java [hunk 2]
```diff
@@ -5,35 +5,51 @@
...
+        
+        if (count > (a.length / 2)) {
+            return Optional.of(candidate);
+        }
+        
+        return Optional.empty();
+    }
+
+
+
```
Comment (vil02) [inline src/main/java/com/thealgorithms/others/BoyerMoore.java:26]: This should fix clang-tidy warning.
```suggestion
```

File: src/main/java/com/thealgorithms/others/BoyerMoore.java [hunk 3]
```diff
@@ -5,35 +5,51 @@
...
 
 public final class BoyerMoore {
     private BoyerMoore() {
     }
 
-    public static int findmajor(final int[] a) {
+    public static Optional<Integer> findMajor(final int[] a) {
+        int candidate = findCandidate(a);
+        int count = countOccurrences(a, candidate);
+
```
Comment (vil02) [inline src/main/java/com/thealgorithms/others/BoyerMoore.java:17]:
```suggestion

```

File: src/main/java/com/thealgorithms/others/BoyerMoore.java [hunk 4]
```diff
@@ -5,35 +5,51 @@
...
 
-    public static int findmajor(final int[] a) {
+    public static Optional<Integer> findMajor(final int[] a) {
+        int candidate = findCandidate(a);
+        int count = countOccurrences(a, candidate);
+        
+        if (count > (a.length / 2)) {
+            return Optional.of(candidate);
+        }
+
```
Comment (vil02) [inline src/main/java/com/thealgorithms/others/BoyerMoore.java:21]:
```suggestion

```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/others/BoyerMoore.java [hunk 5]
```diff
@@ -5,35 +5,51 @@
...
         }
-        count = 0;
-        for (final var j : a) {
-            if (j == cand) {
+        return candidate;
+    }
+
+    private static int countOccurrences(final int[] a, int candidate) {
+        int count = 0;
+        for (final int j : a) {
```
[the remark this answers is not in the document]
Comment (prathameshpowar1910) [reply src/main/java/com/thealgorithms/others/BoyerMoore.java:47]: No it is not necessary

**Review by vil02 — APPROVED** [review]

**Pull request merged**